Source/Core/Runtime/SceneObjects/SingleObjectReference.cs

```diff
@@ -17,7 +17,7 @@ public T Value
             }
         }
 
-        internal override bool AllowMultipleValues => false;
+        internal override int MaxValuesAllowed => 1;
```

Comment:
Reflecting on it, I really can't think of a scenario where you would need a unmodifiable number of object. I can think of plenty where you would reference a list of objects, but nothing where you would need a cap. I may be missing some use cases but I'd opt for staying with the boolean.

Reply:
- How about a multiple choice component where there is only space for a selected amount of items.
- How about a component which wants to implement save guards to have a specific number of objects with a tag (e.g.: 4 car wheels) in the Scene.